src/main/java/com/google/sps/servlets/SaveRecipeServlet.java

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import java.io.IOException;
+import java.lang.SecurityException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Adds recipe to user's list depending on the following parameters:
+ * "listName" = "cookbook", "planner". The list to add to.
+ * "isPublic" = Boolean
+ * "recipeId" = Long
+ *  userToken is the user's Google id_token provided by user-auth.js/getIdToken().
+ * "userToken" = String
+ */
+@WebServlet("/save-recipe")
+public class SaveRecipeServlet extends HttpServlet {
+
+  private static final String COOKBOOK = "cookbook";
+  private static final String PLANNER = "planner";
+  private static final String PUBLIC_KIND = "PublicRecipe";
```

Comment:
Do you want to just have this be PUBLIC_RECIPE instead of PUBLIC_KIND? Same for private below.

Side note but is it possible to have these in a common file that both this and User.java import so you don't have to define COOKBOOK and PLANNER in two places? If not that's OK but it would be cleaner in the future if you ever had to change/add any.

Reply:
I've decided to just write the literals instead, I think it makes more sense and is less redundant.